pkg/service/handlers/v1-kube-api-authn.go

```diff
@@ -135,31 +133,31 @@ func (h *KubeAPIHandlers) v1getAndVerifyUser(accessKey, secretKey string) (*type
 
 		if refresh.Add(refreshPeriod).Before(now) {
 			clusterUserAttribute.NeedsRefresh = true
-			shouldUpdate = true
+			if _, err := h.clusterUserAttribute.Update(clusterUserAttribute); err != nil {
+				return nil, fmt.Errorf("error updating clusterUserAttribute %s: %w", clusterUserAttribute.Name, err)
+			}
 		}
 	}
 
-	func() { // Deliberately using an anonymous function with an early return here to simplify the logic.
-		now = now.Truncate(time.Second) // Use the second precision.
+	func() { // Using an anonymous function with an early return here to simplify the logic.
+		precision := time.Second
+		now = now.Truncate(precision)
 
 		if clusterAuthToken.LastUsedAt != nil {
-			if now.Equal(clusterAuthToken.LastUsedAt.Time.Truncate(time.Second)) {
+			if now.Equal(clusterAuthToken.LastUsedAt.Time.Truncate(precision)) {
 				// Throttle subsecond updates.
 				return
 			}
 		}
 
 		lastUsedAt := metav1.NewTime(now)
 		clusterAuthToken.LastUsedAt = &lastUsedAt
-		shouldUpdate = true
-	}()
 
-	if shouldUpdate {
-		_, err = h.clusterAuthTokens.Update(clusterAuthToken)
-		if err != nil {
-			return nil, fmt.Errorf("error updating clusterAuthToken %s: %w", clusterAuthToken.Name, err)
+		if _, err = h.clusterAuthTokens.Update(clusterAuthToken); err != nil {
```

Comment:
In the future we may want to migrate to wrangler and use `patch` here instead as we do for the local cluster.